XmlFormatter/src/Hotfolder/HotfolderManager.cs

```diff
@@ -109,7 +118,43 @@ private void Watcher_Changed(object sender, FileSystemEventArgs e)
             {
                 return;
             }
-            ConvertFile(hotfolder, e.FullPath);
+            if (tasks.Find((data) => { return data.InputFile == e.FullPath; }) == null)
+            {
+                tasks.Add(new HotfolderTask(e.FullPath, hotfolder));
+            }
+            if (!locked)
+            {
+                locked = true;
+                Task<bool> convertTask = AsyncConvertFiles();
+                convertTask.ContinueWith((result) =>
+                {
+                    locked = false;
+                });
+            }
+        }
+
+        private async Task<bool> AsyncConvertFiles()
+        {
+            List<HotfolderTask> tasksToDo = new List<HotfolderTask>();
+            lock (tasks)
+            {
+                foreach (HotfolderTask task in tasks)
+                {
+                    tasksToDo.Add(task);
+                }
+                tasks.Clear();
+            }
+
+            if (tasksToDo.Count == 0)
+            {
+                return true;
+            }
+
+            foreach (HotfolderTask task in tasksToDo)
+            {
+                ConvertFile(task.Configuration, task.InputFile);
+            }
+            return await AsyncConvertFiles();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add '.ConfigureAwait(false)' to this call to allow execution to continue in any thread.](https://app.codacy.com/manual/simonaberle/XmlFormatter/pullRequest?prid=5090949)